backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/projection/impl/AbstractNotNestableElasticsearchProjection.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.elasticsearch.search.projection.impl;
+
+import java.util.Set;
+
+import org.hibernate.search.backend.elasticsearch.search.common.impl.ElasticsearchSearchIndexScope;
+import org.hibernate.search.engine.reporting.spi.EventContexts;
+import org.hibernate.search.engine.search.projection.spi.SearchProjectionBuilder;
+
+import com.google.gson.JsonObject;
+
+public abstract class AbstractNotNestableElasticsearchProjection<P> implements ElasticsearchSearchProjection<P> {
+
+	protected final Set<String> indexNames;
+
+	protected AbstractNotNestableElasticsearchProjection(ElasticsearchSearchIndexScope<?> scope) {
+		indexNames = scope.hibernateSearchIndexNames();
+	}
+
+	@Override
+	public final Extractor<?, P> request(JsonObject requestBody, ProjectionRequestContext context) {
+		if ( context.absoluteCurrentFieldPath() != null ) {
+			throw log.cannotUseProjectionInNestedContext(
```

Comment:
FWIW you don't need a whole abstract class just for that, you could just add a method `checkNotNested(SearchQueryElementTypeKey projectionKey, String hint)` to `ProjectionRequestContext`, and call it from the relevant projections.

We already have something similar in `org.hibernate.search.backend.elasticsearch.search.projection.impl.ProjectionRequestContext#checkValidField`.